x-pack/plugin/enrich/qa/common/src/main/java/org/elasticsearch/test/enrich/CommonEnrichRestTestCase.java

```diff
@@ -87,6 +96,18 @@ public void testImmutablePolicy() throws IOException {
         assertTrue(exc.getMessage().contains("policy [my_policy] already exists"));
     }
 
+    public void testDeleteExistingPipeline() throws Exception {
+        // lets not delete the pipeline at first, to test the failure
+        setupGenericLifecycleTest(false);
+
+        ResponseException exc = expectThrows(ResponseException.class,
+            () -> client().performRequest(new Request("DELETE", "/_enrich/policy/my_policy")));
+        assertTrue(exc.getMessage().contains("Could not delete policy [my_policy] because a pipeline is referencing it [my_pipeline]"));
+
+        // delete the pipeline so the policies can be deleted
+        client().performRequest(new Request("DELETE", "/_ingest/pipeline/my_pipeline"));
```

Comment:
should you re-retry the delete policy now and verify it is indeed deleted ?

Reply:
the cleanup code does this after every test